pkg/env/prod.go

```diff
@@ -140,8 +140,9 @@ func newProd(ctx context.Context, stop <-chan struct{}, log *logrus.Entry) (*pro
 		return nil, err
 	}
 
-	p.fpCertificateRefresher = newCertificateRefresher(log, 1*time.Hour, p.serviceKeyvault, RPFirstPartySecretName, stop)
-	err = p.fpCertificateRefresher.Start(ctx)
+	p.fpCertificateRefresher = newCertificateRefresher(log, p.serviceKeyvault, RPFirstPartySecretName)
+	ticker := time.NewTicker(1 * time.Hour)
```

Comment:
I think this is the right thing to make ticker mockable, but I would like implementation details (ticker) to stay inside if `refreshingCertificate`  so we don’t have to create a ticker outside and pass into `Start`. Because it feels like we are just shifting the issue up in the calling stack.

I have an idea how to make it mockable, but not sure if it is going to work. I’ll try quick to see if my idea works. I'll post a patch if I manage to implement my idea.